cmake/FindSDL2.cmake

```diff
@@ -0,0 +1,97 @@
+# FindSDL2.cmake
```

Comment:
Does the SDL team provide similar user-facing cmake files? If we can just use theirs then it may be less to maintain.

Okay to do in a follow-up PR.

Reply:
CMake has a built-in `FindSDL.cmake`, but it only works with SDL 1.2.  I believe the MinGW development library comes with some "config" files, but not "finder" scripts (and the config files need to be edited by hand to be useful anyway).

There are other "finder" scripts out there for SDL2, but not one for all three of SDL2, SDL2_mixer, and SDL2_net that were uniform in quality, so I wrote my own.

Reply:
SDL2 comes with `SDL2Config.cmake`, `SDL2ConfigVersion.cmake`, `SDL2Targets-noconfig.cmake`, and `SDL2Targets.cmake`... I don't know how much of a drop-in they are though. Neither mixer nor net come with them either.